### ISSUE TITLE
💅 useSimpleNumberKeys doesn't ignore comments

### ISSUE THREAD

**Issue by yphoenix** [biomejs/biome#918]
### Environment information

```bash
CLI:
  Version:                      1.4.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.2.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/complexity/useSimpleNumberKeys

### Playground link

https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAATQB5AE8AYgBqACAAPQAgAHsACgAJAC8ALwAgAEMAbwBtAG0AZQBuAHQACgAJADQAMgA6ACAAewAKAAkAfQAKAH0ACgA%3D

### Expected result

Should ignore the comments and not flag a lint error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by kalleep** [comment]
Hey :wave: I would like to give this a try
